compiler/src/dotty/tools/dotc/transform/TailRec.scala

```diff
@@ -407,8 +408,12 @@ class TailRec extends MiniPhase {
           assert(false, "We should never have gotten inside a pattern")
           tree
 
-        case _: ValDef | _: DefDef | _: Super | _: This |
-             _: Literal | _: TypeTree | _: TypeDef | EmptyTree =>
+        case tree: DefDef =>
+          noTailTransform(tree.rhs)
+          tree
+
+        case _: ValDef | _: Super | _: This | _: Literal |
```

Comment:
Why aren't ValDefs traversed ?

Reply:
This is not new code. My guess is that they can’t be in tail position. But we should probably traverse them for the same reasons we want to traverse local DefDef